lucetc/src/lib.rs

```diff
@@ -150,7 +155,17 @@ impl Lucetc {
     pub fn new<P: AsRef<Path>>(input: P) -> Self {
         let input = input.as_ref();
         Self {
-            input: input.to_owned(),
+            input: LucetcInput::Path(input.to_owned()),
```

Comment:
here's the only change in `new`, wrapping the path in the corresponding enum variant.